src/GitHub/App/Token/Permissions.hs

```diff
@@ -0,0 +1,306 @@
+-- | Type-safe implementation for requesting 'AccessToken' permissions
+--
+-- <https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#create-an-installation-access-token-for-an-app>
+--
+-- Usage:
+--
+-- 1. Use a constructor function for a specific permission, each of which only
+--    accepts appropriate values (e.g. you can't ask for @admin@ on a permission
+--    that only allows @read@ or @write@).
+-- 2. Combine these using 'Permissions' 'Semigroup' instance.
+--
+-- For example:
+--
+-- @
+-- let permissions = 'actions' 'Read' <> 'checks' 'Write'
+--
+-- 'generateInstallationTokenScoped' AllRepositories permissions creds installationId
+-- @
+--
+-- Supplying the same permission more than once will take the higher:
+--
+-- @
+-- 'checks' 'Read' <> 'checks' 'Write' == 'checks' 'Write'
+-- @
+module GitHub.App.Token.Permissions
+  ( Permissions
+  , Read (..)
+  , Write (..)
+  , Admin (..)
+  , actions
+  , administration
+  , checks
+  , codespaces
+  , contents
+  , dependabot_secrets
+  , deployments
+  , environments
+  , issues
+  , metadata
+  , packages
+  , pages
+  , pull_requests
+  , repository_custom_properties
+  , repository_hooks
+  , repository_projects
+  , secret_scanning_alerts
+  , secrets
+  , security_events
+  , single_file
+  , statuses
+  , vulnerability_alerts
+  , workflows
+  , members
+  , organization_administration
+  , organization_custom_roles
+  , organization_custom_org_roles
+  , organization_custom_properties
+  , organization_copilot_seat_management
+  , organization_announcement_banners
+  , organization_events
+  , organization_hooks
+  , organization_personal_access_tokens
+  , organization_personal_access_token_requests
+  , organization_plan
+  , organization_projects
+  , organization_packages
+  , organization_secrets
+  , organization_self_hosted_runners
+  , organization_user_blocking
+  , team_discussions
+  , email_addresses
+  , followers
+  , git_ssh_keys
+  , gpg_keys
+  , interaction_limits
+  , profile
+  , starring
+  ) where
+
+import GitHub.App.Token.Prelude hiding (Read)
+
+import Data.Aeson (ToJSON (..))
+import Data.Map.Monoidal.Strict (MonoidalMap)
+import Data.Map.Monoidal.Strict qualified as MonoidalMap
+
+{-# ANN module ("HLint: ignore Use camelCase" :: String) #-}
```

Comment:
Or I could make the constructor functions (e.g.) `pullRequests`. They don't have to match the object keys.